akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphMergeLatestSpec.scala

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.scaladsl
+
+import akka.stream._
+import akka.stream.testkit.Utils._
+import akka.stream.testkit._
+import akka.stream.testkit.scaladsl.TestSource
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class GraphMergeLatestSpec extends TwoStreamsSetup {
+  import GraphDSL.Implicits._
+
+  override type Outputs = List[Int]
+
+  override def fixture(b: GraphDSL.Builder[_]): Fixture = new Fixture(b) {
+    val merge = b add MergeLatest[Int](2)
+
+    override def left: Inlet[Int] = merge.in(0)
+    override def right: Inlet[Int] = merge.in(1)
+    override def out: Outlet[Outputs] = merge.out
+
+  }
+
+  "mergeLatest" must {
+
+    "start emit values only after each input stream emitted value" in assertAllStagesStopped {
+      val up1 = TestSource.probe[Int]
+      val up2 = TestSource.probe[Int]
+      val up3 = TestSource.probe[Int]
+      val probe = TestSubscriber.manualProbe[List[Int]]()
+
+      val (in1, in2, in3) = RunnableGraph.fromGraph(GraphDSL.create(up1, up2, up3)((_, _, _)) { implicit b ⇒ (s1, s2, s3) ⇒
+        val m = b.add(MergeLatest[Int](3))
+
+        s1 ~> m.in(0)
+        s2 ~> m.in(1)
+        s3 ~> m.in(2)
+        m.out ~> Sink.fromSubscriber(probe)
+        ClosedShape
+      }).run()
+
+      val subscription = probe.expectSubscription()
+
+      subscription.request(1)
+      probe.expectNoMessage(10.millis)
+      in1.sendNext(1)
+      probe.expectNoMessage(10.millis)
+      in2.sendNext(2)
+      probe.expectNoMessage(10.millis)
+      in3.sendNext(3)
+      probe.expectNext(List(1, 2, 3))
```

Comment:
This is not any different semantics than a normal merge though so far :)
Test should be:

```
in1.sendNext(1)
in2.sendNext(2)
in2.sendNext(22)
in3.sendNext(3)
in3.sendNext(33)
in3.sendNext(333)
```

Which should then when pulled end up with `List(1, 22, 333)` those are the "Latest" values

Reply:
Nevermind, I just saw you did such test in the next testcase. Sorry for the noise

Reply:
Reminding the image as well:

![](https://blogs.endjin.com/wp-content/uploads/2014/05/combine-latest.png)

green 1 is sent
yellow 2 is sent
result 1 2
yellow 22 is sent
result 1 22
green 11 is sent
result 11 22

Reply:
Nevermind, I see you added a test for this in the below testcase :) Sorry for the noise